drivers: Rename renesas ra drivers

aaillet opened the pull request:
With current name nomenclature for Renesas RA driver, it's not possible to target them correctly in the MAINTAINER file.

Rename all drivers to be able to delimit Renesas areas correctly.

Blocking #67487

aaillet committed:
drivers: clock: Rename renesas ra driver

Need this rename to be able to target this driver
in the "Renesas RA" maintainer area.

Signed-off-by: Aymeric Aillet <aymeric.aillet@iot.bzh>

aaillet committed:
drivers: gpio: Rename renesas ra driver

Need this rename to be able to target this driver
in the "Renesas RA" maintainer area.

Signed-off-by: Aymeric Aillet <aymeric.aillet@iot.bzh>

aaillet committed:
drivers: intc: Rename renesas ra driver

Need this rename to be able to target this driver
in the "Renesas RA" maintainer area.

Signed-off-by: Aymeric Aillet <aymeric.aillet@iot.bzh>

aaillet committed:
drivers: pinctrl: Rename renesas ra driver

Need this rename to be able to target this driver
in the "Renesas RA" maintainer area.

Signed-off-by: Aymeric Aillet <aymeric.aillet@iot.bzh>

aaillet committed:
drivers: serial: Rename renesas ra driver

Need this rename to be able to target this driver
in the "Renesas RA" maintainer area.

Signed-off-by: Aymeric Aillet <aymeric.aillet@iot.bzh>

henrikbrixandersen reviewed (changes requested):
The Kconfig renames should be added to the migration guide/release notes.

aaillet committed:
doc: releases: 3.6: note on renaming RA drivers kconfig options

Add a note about the renaming of several Renesas RA driver kconfig options.
These options have been renamed when renaming several driver file names.

Signed-off-by: Aymeric Aillet <aymeric.aillet@iot.bzh>

aaillet committed:
doc: releases: migration-guide: 3.6: note on renaming RA kconfig options

Add a note about the renaming of several Renesas RA driver kconfig options.
These options have been renamed when renaming several driver file names.

Signed-off-by: Aymeric Aillet <aymeric.aillet@iot.bzh>

aaillet commented:
> The Kconfig renames should be added to the migration guide/release notes.

I added notes to both release note and migration guide, please @henrikbrixandersen tell me if I did it well !
Thanks

kartben commented on doc/releases/migration-guide-3.6.rst:
```diff
@@ -271,6 +271,13 @@ Device Drivers and Device Tree
 
+* Several Renesas RA series drivers Kconfig options have been renamed:
+
+  * :kconfig:option:`CLOCK_CONTROL_RA` -> :kconfig:option:`CLOCK_CONTROL_RENESAS_RA`
+  * :kconfig:option:`GPIO_RA` -> :kconfig:option:`GPIO_RENESAS_RA`
+  * :kconfig:option:`PINCTRL_RA` -> :kconfig:option:`PINCTRL_RENESAS_RA`
+  * :kconfig:option:`UART_RA` -> :kconfig:option:`UART_RENESAS_RA`
```
```suggestion
  * ``CLOCK_CONTROL_RA`` -> :kconfig:option:`CLOCK_CONTROL_RENESAS_RA`
  * ``GPIO_RA`` -> :kconfig:option:`GPIO_RENESAS_RA`
  * ``PINCTRL_RA`` -> :kconfig:option:`PINCTRL_RENESAS_RA`
  * ``UART_RA`` -> :kconfig:option:`UART_RENESAS_RA`
```

kartben commented on doc/releases/release-notes-3.6.rst:
```diff
@@ -157,6 +157,7 @@ Drivers and Sensors
 * Clock control
 
   * Renesas R-Car clock control driver now supports Gen4 SoCs
+  * Renamed :kconfig:option:`CLOCK_CONTROL_RA` with :kconfig:option:`CLOCK_CONTROL_RENESAS_RA`
```
```suggestion
  * Renamed ``CLOCK_CONTROL_RA`` with :kconfig:option:`CLOCK_CONTROL_RENESAS_RA`
```

kartben commented on doc/releases/release-notes-3.6.rst:
```diff
@@ -179,6 +180,7 @@ Drivers and Sensors
 * GPIO
 
   * Renesas R-Car GPIO driver now supports Gen4 SoCs
+  * Renamed :kconfig:option:`GPIO_RA` with :kconfig:option:`GPIO_RENESAS_RA`
```
```suggestion
  * Renamed ``GPIO_RA`` with :kconfig:option:`GPIO_RENESAS_RA`
```

kartben commented on doc/releases/release-notes-3.6.rst:
```diff
@@ -204,6 +206,7 @@ Drivers and Sensors
 * Pin control
 
   * Renesas R-Car pinctrl driver now supports Gen4 SoCs
+  * Renamed :kconfig:option:`PINCTRL_RA` with :kconfig:option:`PINCTRL_RENESAS_RA`
```
```suggestion
  * Renamed ``PINCTRL_RA`` with :kconfig:option:`PINCTRL_RENESAS_RA`
```

kartben commented on doc/releases/release-notes-3.6.rst:
```diff
@@ -225,6 +228,8 @@ Drivers and Sensors
 
 * Serial
 
+  * Renamed :kconfig:option:`UART_RA` with :kconfig:option:`UART_RENESAS_RA`
```
```suggestion
  * Renamed ``UART_RA`` with :kconfig:option:`UART_RENESAS_RA`
```

aaillet commented:
Addressed @kartben request

aaillet commented on doc/releases/migration-guide-3.6.rst:
```diff
@@ -271,6 +271,13 @@ Device Drivers and Device Tree
 
+* Several Renesas RA series drivers Kconfig options have been renamed:
+
+  * ``CLOCK_CONTROL_RA`` -> :kconfig:option:`CLOCK_CONTROL_RENESAS_RA`
```
Thanks,
Don't I need the "config" prefix for previous names ?

henrikbrixandersen replied:
> Thanks, Don't I need the "config" prefix for previous names ? It feels weird for me not to use the same name nomenclature for old and new names

Yes, you need the `CONFIG_` prefix.

henrikbrixandersen replied:
> Just for you to know, I based these changed on https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/releases/release-notes-3.6.rst?plain=1#L361 So it looks like these rules are not widely known :)

Yeah, that change is wrong. Surprised it builds.

aaillet replied:
Thanks, Added config prefix.
If I'm understanding well, `:kconfig:option:` is checking for an existing Kconfig option at doc build time, that's it ?

henrikbrixandersen replied:
It's creating a link to it, yes - and we cannot create links to the now non-existing, previous names.

aaillet commented:
Added "CONFIG" prefix to previous and new Kconfig options names in release note and migration guide

aaillet commented on doc/releases/release-notes-3.6.rst:
```diff
@@ -157,6 +157,7 @@ Drivers and Sensors
 * Clock control
 
   * Renesas R-Car clock control driver now supports Gen4 SoCs
+  * Renamed ``CONFIG_CLOCK_CONTROL_RA`` with :kconfig:option:`CONFIG_CLOCK_CONTROL_RENESAS_RA`
```
Fixed !
